src/snowflake/snowpark/session.py

```diff
@@ -974,12 +1007,143 @@ def get_req_identifiers_list(
 
             return res
 
-        # always include cloudpickle
+        # Add dependency packages
+        if dependency_packages:
+            for package in dependency_packages:
+                name = package.name
+                version = package.specs[0][1] if package.specs else None
+
+                # Add to packages dictionary
+                if name in result_dict:
+                    if result_dict[name] != str(package):
+                        raise ValueError(
+                            f"Cannot add dependency package '{name}'{'( version '+version+')' if version else ''} "
+                            f"because {result_dict[name]} is already added."
+                        )
+                else:
+                    result_dict[name] = str(package)
+
+        # Add the Snowpark package to packages dict, based on client's environment (or latest), if not already added
+        # add_snowpark_package(result_dict, valid_packages)
```

Comment:
should this be uncommented?

Reply:
Not adding snowpark python package automatically in this PR (moved to next PR for simplicity).

This is because adding snowpark python package only makes sense for sprocs and not udfs (this distinction was not made in this PR)

Reply:
okay. Can you remove the comment here then? Re-add it in the next PR